src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -237,21 +245,26 @@ public SSLHelper setApplicationProtocols(List<HttpVersion> applicationProtocols)
     If you don't specify a key store, and don't specify a system property no key store will be used
     You can override this by specifying the javax.echo.ssl.keyStore system property
      */
-  private SslContext createContext(VertxInternal vertx) {
+  private SslContext createContext(VertxInternal vertx, X509KeyManager mgr) {
     try {
-      KeyManagerFactory keyMgrFactory = getKeyMgrFactory(vertx);
       TrustManagerFactory trustMgrFactory = getTrustMgrFactory(vertx);
       SslContextBuilder builder;
       if (client) {
         builder = SslContextBuilder.forClient();
+        KeyManagerFactory keyMgrFactory = getKeyMgrFactory(vertx);
         if (keyMgrFactory != null) {
           builder.keyManager(keyMgrFactory);
         }
       } else {
-        if (keyMgrFactory == null) {
-          throw new VertxException("Key/certificate is mandatory for SSL");
+        if (mgr != null) {
+          builder = SslContextBuilder.forServer(mgr.getPrivateKey(null), "wibble", mgr.getCertificateChain(null));
```

Comment:
Is `wibble` a magic value required here? Or would it be possible to use other passwords too =)